source/Interfaces/NetWorth/NetworkInterface.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * This file is part of the Fall 2023, CSE 491 course project.
+ * @brief A networking interface that allows information to be sent across a  network
+ * @note Status: PROTOTYPE
+ **/
+
+#pragma once
+
+#include <cassert>
+#include <string>
+#include <vector>
+#include <SFML/Network/UdpSocket.hpp>
+#include <SFML/Network/Packet.hpp>
+
+#include "../../core/InterfaceBase.hpp"
+
+using namespace sf;
+
+namespace cse491 {
+    namespace netWorth{
+    /**
+     * TODO: Delete or incorporate
+     */
+        class NetworkingInterface{
+        private:
+
+        protected:
+            UdpSocket mSocket; ///The socket we are going to make our connection
+            IpAddress mLocalAddress; ///the local address of the server
+            //Thought about making mClients a shared pointer to a vector, but it'll be a vector for now
+            std::vector<std::string> mClients; ///list of all the clients that will connect with the server
+            unsigned short mPort;
+
+        public:
+            NetworkingInterface() = default;
+            ~NetworkingInterface() = default;
+            /**
+             * Receives a socket that has been connected between client and server
+             * @return the udp socket
+             */
+            UdpSocket * GetSocket(){
+                return &mSocket;
+            }
+//            virtual Packet CreateActionPacket(std::string action){
+//                //Creating packet to send to server of actions
+//                Packet packet;
+//                size_t action_num = action_map[action];
+//                packet << static_cast<int>(action_num);
+//                return packet;
+//            }
+
+            /**
+             * Sends a packet across the socket
+             * @param packet the packet we want to send
+             * @param destAddr the destination address we want to send to
+             * @param port the port of the connection
+             */
+            virtual void SendPacket(Packet packet, IpAddress destAddr, const unsigned short port){
+                mSocket.send(packet, destAddr, port);
+            }
+            /**
+             * Starts the connection by receiving the first packet
+             * @return the IP of the server
+             */
+            virtual IpAddress ReceivePacket(){
+                char buffer[1024];
```

Comment:
I'd recommend changing this over to a standard library array.  Specifically, it would look like:

```
std::array<char, 1024> buffer;
```

It would otherwise behave the same, but would have some member function to help out, like `buffer.size()`.

Reply:
The code used here was given from the SFML documentation and changing the type to std::array<char, 1024> does not match the type of a parameter for a socket receive function that the SFML API uses. Not sure what to do to change this other than doing some sort of potential strange type-casting.

Reply:
If you use std::array, you can always call `.data()` on it to convert it back to a regular array.  Either way should be fine though.

Reply:
Ok, yeah that seems to work for now, we aren't actually using that class at the moment but we will make the change so that when our class hierarchy is complete we have that additional functionality.